src/beliefpropagation.jl

```diff
@@ -109,27 +137,45 @@ function get_environment(tn::ITensorNetwork, mts::DataGraph, verts::Vector; dir=
     return get_environment(tn, mts, setdiff(vertices(tn), verts))
   end
 
-  env_tensors = ITensor[mts[e] for e in boundary_edges(mts, subgraphs; dir=:in)]
-  return vcat(
-    env_tensors,
-    ITensor[tn[v] for v in setdiff(flatten([vertices(mts[s]) for s in subgraphs]), verts)],
-  )
+  env_tns = ITensorNetwork[mts[e] for e in boundary_edges(mts, subgraphs; dir=:in)]
+  central_tn = ITensorNetwork([
+    tn[v] for v in setdiff(flatten([vertices(mts[s]) for s in subgraphs]), verts)
+  ])
+  return vcat(env_tns, ITensorNetwork[central_tn])
+end
+
+function get_environment(
+  output_type::Type, tn::ITensorNetwork, mts::DataGraph, verts::Vector; kwargs...
+)
+  itns = get_environment(tn::ITensorNetwork, mts::DataGraph, verts::Vector; kwargs...)
+
+  if output_type == Vector{ITensorNetwork}
+    return itns
+  else
+    itn = reduce(⊗, itns)
+    if output_type == ITensorNetwork
+      return itn
+    elseif output_type == Vector{ITensor}
+      return ITensor[itn[v] for v in vertices(itn)]
+    else
+      error("Output Type for get_environment not Supported!")
+    end
+  end
 end
 
 """
 Calculate the contraction of a tensor network centred on the vertices verts. Using message tensors.
 Defaults to using tn[verts] as the local network but can be overriden
 """
-function calculate_contraction(
+function calculate_contraction_network(
```

Comment:
What about `approx_subnetwork` or `approx_network_region`, as in it outputs an approximation of a subnetwork or region of the network?